openshift-hack/cmd/k8s-tests-ext/k8s-tests.go

```diff
@@ -88,10 +88,29 @@ func main() {
 		Qualifiers: []string{withExcludedTestsFilter(`name.contains('[Serial]')`)},
 	})
 
-	for k, v := range image.GetOriginalImageConfigs() {
+	originals := image.GetOriginalImageConfigs()
+	originalImages := make(map[int]e.Image)
+	for k, v := range originals {
 		image := convertToImage(v)
 		image.Index = int(k)
-		kubeTestsExtension.RegisterImage(image)
+		originalImages[image.Index] = image
+	}
+
+	mirror := "quay.io/openshift/community-e2e-images"
+	if v := os.Getenv("TEST_IMAGE_MIRROR"); len(v) > 0 {
+		mirror = v
+	}
+
+	mapped := image.GetMappedImageConfigs(originals, mirror)
+	for k, v := range mapped {
+		original, ok := originalImages[int(k)]
+		if !ok {
+			continue
+		}
+		image := convertToImage(v)
+		image.Index = int(k)
+		original.Mapped = &image
+		kubeTestsExtension.RegisterImage(original)
```

Comment:
What about just iterating over the originals once, and we go, build the mapped image?

```
	originals := image.GetOriginalImageConfigs()
	mapped := image.GetMappedImageConfigs(originals, mirror)

	for k, v := range originals {
		mappedImage := convertToImage(mapped[k], k)
		image := convertToImage(v, k)
		image.Mapped = &mappedImage
		kubeTestsExtension.RegisterImage(image)
	}
```